### ISSUE TITLE
Examine integration of Auth module on Supabase with the backend

### ISSUE THREAD

**Issue by shaylavi** [openheartmind/WISDOM-Backend-MVP#12]

**Comment by dipanshuhappy** [comment]
@shaylavi  The backend codebase currently doesn't seem to have typescript support , are you adding that or should i check it ?

**Comment by shaylavi** [comment]
@dipanshuhappy that's correct thanks. I haven't finished setting it up properly as I'm struggling with the Azure deployment. I added a comment about it in the ticket for Ben but we can screenshare later if you might able to help. I might need to recreate this code base for a clean deployment without all the test commits I pushed

**Comment by dipanshuhappy** [comment]
> [@dipanshuhappy](https://github.com/dipanshuhappy) that's correct thanks. I haven't finished setting it up properly as I'm struggling with the Azure deployment. I added a comment about it in the ticket for Ben but we can screenshare later if you might able to help. I might need to recreate this code base for a clean deployment without all the test commits I pushed

Ah okay makes sense. Thanks

**Comment by smuriu** [comment]
@shaylavi I tried a moleculer project with the `project-typescript` template which set up a functional moleculer project with typescript support. I also started on a service for the supabase client integration. However, was not able to have the types for the supabase service auto-discovered by other moleculer micro-services. So in `api` service, I added my supabase service as a mixin. But my supabase service methods seem unknown (type-wise) from the `api` service

**Comment by smuriu** [comment]
Hey @shaylavi, @dipanshuhappy , I added a supabase service along with the initial schema in my latest commit (b74aae427ffc1f3ea26e597685d67270c557f2cc). In this version, the supabase service is a "full" service with it's own actions. I'm having trouble extracting it to a re-usable mixin with the typescript support; but it seems to work well as a micro-service (not goal). Any help would be appreciated

**Comment by dipanshuhappy** [comment]
> [@shaylavi](https://github.com/shaylavi) I tried a moleculer project with the `project-typescript` template which set up a functional moleculer project with typescript support. I also started on a service for the supabase client integration. However, was not able to have the types for the supabase service auto-discovered by other moleculer micro-services. So in `api` service, I added my supabase service as a mixin. But my supabase service methods seem unknown (type-wise) from the `api` service

So you are trying to add supabase to the molecular ?  I think @shaylavi  is working on adding typescript .. i can also experiment on it rather than waiting to speed things up

**Comment by dipanshuhappy** [comment]
> Hey [@shaylavi](https://github.com/shaylavi), [@dipanshuhappy](https://github.com/dipanshuhappy) , I added a supabase service along with the initial schema in my latest commit ([b74aae4](https://github.com/openheartmind/WISDOM-Backend-MVP/commit/b74aae427ffc1f3ea26e597685d67270c557f2cc)). In this version, the supabase service is a "full" service with it's own actions. I'm having trouble extracting it to a re-usable mixin with the typescript support; but it seems to work well as a micro-service (not goal). Any help would be appreciated

Hmm okay i will check it , tomorrow morning and let you know

**Comment by shaylavi** [comment]
Thanks guys, I've pushed a new version of the repo with TS embedded, but I'm not sure it solves the missing types of Supabase.

I'm still stuck with Azure and it's a real pain. As Ben mentioned, it's the container that wraps Azure's web app service that for some reason breaks down completely when something isn't right and I couldn't find a way to recover it expect recreate the whole service. it's a pain and I'm still investigating.

If we go with an actual VM it would be easier but the price difference is instant and I'd like to believe there should be a fairly simple solution to use the web app service

**Comment by dipanshuhappy** [comment]
> Hey [@shaylavi](https://github.com/shaylavi), [@dipanshuhappy](https://github.com/dipanshuhappy) , I added a supabase service along with the initial schema in my latest commit ([b74aae4](https://github.com/openheartmind/WISDOM-Backend-MVP/commit/b74aae427ffc1f3ea26e597685d67270c557f2cc)). In this version, the supabase service is a "full" service with it's own actions. I'm having trouble extracting it to a re-usable mixin with the typescript support; but it seems to work well as a micro-service (not goal). Any help would be appreciated

So apparently every service can be used as a mixin , unless we want to customize the behaviour of the mixin based on the service .. then it makes since to use a supabase.mixin.ts rather than a supabase.service.ts

**Comment by shaylavi** [comment]
so how about I'll restart the code base for a plain JS template? it'll simplify the deployment scripts I reckon

**Comment by dipanshuhappy** [comment]
> so how about I'll restart the code base for a plain JS template? it'll simplify the deployment scripts I reckon

Nah no need to that .. i am just saying i don't think we need the auth module as a mixin .. and we are kind of good to go in closing this because @smuriu  has made a basic email and password supabase auth module .
